rtt_tf/rtt_tf-component.cpp

```diff
@@ -80,7 +81,7 @@ namespace rtt_tf
   {
     tf2_msgs::TFMessage msg;
     // resolve names and copy transforms to message
-    msg.transforms.reserve(tforms.size());
+    msg.transforms.resize(tforms.size());
     std::transform(tforms.begin(), tforms.end(), msg.transforms.begin(), PrefixResolver(prefix));
```

Comment:
But if the elements are `push_back()`-ed into the `msg.transforms` vector, this would keep growing instead of substituting the elements in there, right? I don't clearly see the advantage.
This change was to solve the issue of `.reserve()` not actually initializing that memory and affecting `size()` but only the allocation and `capacity()` value. This caused a segfault before.

In other words, what happens if the new set of transforms is smaller in size than the previous one? Would it override only the first `tforms.size()` elements of `msg.transforms` or append to `msg.transforms` increasing its `size()` every time? What we want is that if fills an exact size of the vector.

Reply:
I meant that one could use the `back_inserter` to `push_back()` elements after `reserve()`-ing instead of `resize()`-ing as you originally wrote. But indeed, all you would achieve is make sure that the heap allocation takes place at start, so there's no real advantage in this scenario.